server/auth/auth.go

```diff
@@ -918,12 +933,33 @@ func (auth *AuthManager) handleConnect(conn comms.Link, msg *msgjson.Message) *m
 
 	// Send pending requests for this user.
 	for _, pr := range pendingReqs {
+		// Certain match-related messages should only be sent if the match is
+		// still active.
+		mid, err := pr.req.ExtractMatchID()
+		if err != nil {
+			log.Errorf("Failed to read matchid field from '%s' payload: %v", pr.req.Route, err)
+			// just send it
+		} else if mid != nil && pr.req.Route != msgjson.RevokeMatchRoute {
+			// Always send revoke_match, but only send others with matchid set
+			// if the match is still active.
+			var matchID order.MatchID
+			copy(matchID[:], mid)
+			if !activeMatchIDs[matchID] {
+				log.Debugf("Not sending pending '%s' request to %v for inactive match %v.",
+					pr.req.Route, user, matchID)
+				continue // don't send this request
+			}
+		}
+
 		log.Debugf("Sending pending '%s' request to user %v: %v", pr.req.Route, pr.user, pr.req.String())
 		// Use the AuthManager method to send so that failed requests, which
 		// result in client removal, will go back into the client's pending
 		// requests. Subsequent requests that follow removal also fail and go
 		// back into the pending requests.
 		connectTimeout := DefaultConnectTimeout
+		if pr.req.Route == msgjson.RevokeMatchRoute {
+			connectTimeout = 4 * time.Hour // a little extra time for revoke_match to be courteous
```

Comment:
It's little cost to queue thousands of these for hours, and it helps the client when they do reconnect, especially if their dexc never restarted, only reconnected after prolonged outage (laptop suspend or network off).